### PR TITLE
Bugfix/Select setting improvements

### PR DESCRIPTION
**Pull request by Sisha3342**

### CODE REVIEW

**Comment by yadamskaya** [comment]
@Sisha3342 I made some review and there are some things:
1. We have the blue checkbox of bool setting and blue circle of slider setting. I think that it would be better to color them too, at least because the arrow in the selector setting was changed.
In case `light theme` - dark color for checkbox and circle
In case `dark theme` - white color for both checkbox and circle
Let me know if you don't think so or mb have other idea. @exadel-inc/ui-playground-team 
2. Another interesting thing appears in case when Playground inside Playground. 
When Playground outside has a `dark theme` the list of select items of Playground inside is also dark. But it must not change.
And when Playground inside has a `dark theme` the list of select items of this Playground is white. It must be dark.
When both Playgrounds have `dark theme` it works correctly! :)

File: src/plugins/settings/setting.ts [hunk 1]
```diff
@@ -18,7 +18,9 @@ export abstract class UIPSetting extends ESLBaseElement {
   /** [Target's]{@link target} attribute which is changed by setting. */
   @attr() public attribute: string;
   /** Target to which setting's changes are attached. */
-  @attr() public target: string;
+  public get target(): string {
+    return this.getAttribute('target') || this.$settings.target;
+  }
```
Comment (ala-n) [inline src/plugins/settings/setting.ts:23]: Can't we change it from JS?
Why do we commit the setter part of the property for this attribute?

**Review by ala-n — APPROVED** [review]

**Review by yadamskaya — APPROVED** [review]

File: src/settings/select-setting/select-setting.ts [hunk 2]
```diff
@@ -27,44 +32,52 @@ export class UIPSelectSetting extends UIPSetting {
...
     select.id = `${UIPSelectSetting.is}-${randUID()}`;
-
     this.querySelectorAll('option').forEach(option => select.add(option));
+    return select;
+  }
+
+  @memoize()
+  protected get root(): UIPRoot {
+    return this.closest(UIPRoot.is) as UIPRoot;
+  }
```
Comment (nattallius) [inline src/settings/select-setting/select-setting.ts:71]: does it used anywhere?

File: src/settings/select-setting/select-setting.ts [hunk 3]
```diff
@@ -149,6 +162,19 @@ export class UIPSelectSetting extends UIPSetting {
...
+  protected clearInconsistency(): void {
+    this.select.remove(this.settingOptions.indexOf(UIPSelectSetting.inconsistentValue));
+  }
+
+  @listen({event: 'uip:configchange', target: '::parent(.uip-root)'})
+  protected onRootThemeChange(e: CustomEvent): void {
+    if (e.detail.attribute !== 'dark-theme') return;
+    let dropdownClass = UIPSelectSetting.dropdownClass;
+    if (e.detail.value !== null) dropdownClass += ' uip-dark-dropdown';
+    this.$field.setAttribute('dropdown-class', dropdownClass);
```
Comment (nattallius) [inline src/settings/select-setting/select-setting.ts:175]:
```suggestion
this.$field.dropdownClass = dropdownClass;
```

**Review by yadamskaya — APPROVED** [review]

**Review by nattallius — APPROVED** [review]

**Review by AlexanderBazukevich — APPROVED** [review]

**Review by PolinaVolchetskaya — APPROVED** [review]

**Pull request merged**